sklearn/linear_model/_cd_fast.pyx

```diff
@@ -171,7 +171,7 @@ def enet_coordinate_descent(
         # tol *= np.dot(y, y)
         tol *= _dot(n_samples, &y[0], 1, &y[0], 1)
 
-        for n_iter in range(max_iter):
+        for n_iter in range(<Py_ssize_t>max_iter):
```

Comment:
I think it is preferable to change the type of `n_iter` to the one of `max_iter` (i.e. int) instead of casting `max_iter` to `Py_ssize_t` here.

Reply:
As for SGD, `max_iter` will be validated by the Python class to be strictly positive.
In terms of semantic, it is more correct to use unsigned and the casting will be safe.
It can be done once for all in the signature of the function.